handlers/upload.go

```diff
@@ -76,7 +77,46 @@ func (s Server) entryPut() http.HandlerFunc {
 			return
 		}
 
-		metadata, err := s.entryMetadataFromRequest(r)
+		// We allow changing filename, expiration, note, and optionally passphrase.
+		// Passphrase change is specified with either passphrase (string) to set/update,
+		// or passphraseAction:"clear" to remove.
```

Comment:
We shouldn't support two parallel ways of the client indicating the same thing. An empty string passphrase should indicate removing a passphrase.